src/luasocket.c

```diff
@@ -30,6 +30,9 @@
 #include "inet.h"
 #include "tcp.h"
 #include "udp.h"
+#ifndef NETLINK_H
```

Comment:
it's not enough to put ifdef only here.. you should apply it to all code that depends on netlink. also, I don't think NETLINK_H is right here.. you should use convention used by luasocket to other platform-dependent code..